idea/src/org/jetbrains/kotlin/idea/inspections/UnnecessaryVariableInspection.kt

```diff
@@ -27,22 +29,32 @@ import org.jetbrains.kotlin.idea.caches.resolve.analyze
 import org.jetbrains.kotlin.idea.caches.resolve.resolveToCall
 import org.jetbrains.kotlin.idea.core.NewDeclarationNameValidator
 import org.jetbrains.kotlin.idea.refactoring.inline.KotlinInlineValHandler
+import org.jetbrains.kotlin.idea.util.getLineCount
 import org.jetbrains.kotlin.idea.util.nameIdentifierTextRangeInThis
 import org.jetbrains.kotlin.lexer.KtTokens
 import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.anyDescendantOfType
 import org.jetbrains.kotlin.psi.psiUtil.getNextSiblingIgnoringWhitespaceAndComments
 import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
 import org.jetbrains.kotlin.resolve.BindingContext.DECLARATION_TO_DESCRIPTOR
 import org.jetbrains.kotlin.resolve.BindingContext.REFERENCE_TARGET
 import org.jetbrains.kotlin.resolve.DescriptorToSourceUtils
 import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
 class UnnecessaryVariableInspection : AbstractApplicabilityBasedInspection<KtProperty>(KtProperty::class.java) {
 
     override fun isApplicable(element: KtProperty) = statusFor(element) != null
 
     override fun inspectionHighlightRangeInElement(element: KtProperty) = element.nameIdentifierTextRangeInThis()
 
+    override fun inspectionHighlightType(element: KtProperty): ProblemHighlightType {
+        val hasMultiLineBlock = element.initializer?.anyDescendantOfType<LeafPsiElement> { e ->
+            e.node.elementType == KtTokens.LBRACE && e.parent.safeAs<KtExpression>()?.getLineCount()?.let { it > 1 } == true
```

Comment:
I think it might be better to look for `KtBlockExpression` elements instead.

Reply:
KtWhenExpression and KtLambdaArgument have no KtBlockExpression.

https://github.com/JetBrains/kotlin/blob/8884a451ea412b78cb52098669661d8123444c84/idea/testData/inspectionsLocal/unnecessaryVariable/when.kt

https://github.com/JetBrains/kotlin/blob/8884a451ea412b78cb52098669661d8123444c84/idea/testData/inspectionsLocal/unnecessaryVariable/lambda2.kt

Reply:
Thank you.
Can I ask you to introduce smaller functions for readability improvement, please? E.g.
``` kotlin
private fun LeafPsiElement.startsMultilineBlock(): Boolean =
    node.elementType == KtTokens.LBRACE && parent.safeAs<KtExpression>()?.getLineCount()?.let { it > 1 } == true

private fun KtExpression.hasMultiLineBlock(): Boolean = anyDescendantOfType(LeafPsiElement::startsMultilineBlock)
...

val hasMultiLineBlock = element.initializer?.hasMultiLineBlock() == true
```